// spotify-web-api-ts/cjs/apis/ShowsApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
class ShowsApi {
    constructor(http) {
        this.http = http;
    }
    getShow(showId, options) {
        return this.http.get(`/shows/${showId}`, options && { params: options });
    }
    getShowEpisodes(showId, options) {
        return this.http.get(`/shows/${showId}/episodes`, options && { params: options });
    }
    getShows(showIds, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.http.get('/shows', {
                params: Object.assign(Object.assign({}, options), { ids: showIds }),
            });
            return response.shows;
        });
    }
}
exports.ShowsApi = ShowsApi;
